vApp properties not being set with OVF deployment with provider 1.24.1 and 1.24.2

### Terraform Version: 0.13.4

### vSphere Provider Version: 1.24.2


### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {
    vsphere = {
      source  = "hashicorp/vsphere"
      version = "1.24.2"
    }
  }
}

variable "vsphere_user" {
  type = string
}

variable "vsphere_password" {
  type = string
}

variable "vsphere_server" {
  type = string
}

variable "vsphere_insecure" {
  type = bool
  default = true
}

variable "vsphere_datacenter" {
  type = string
}

variable "vsphere_datastore" {
  type = string
}

variable "vsphere_resource_pool" {
  type = string
}

variable "vsphere_host_system" {
  type = string
}

variable "vsphere_network" {
  type = string
}

variable "vsphere_ovf_url" {
  type = string
}

provider "vsphere" {
  user                  = var.vsphere_user
  password              = var.vsphere_password
  vsphere_server        = var.vsphere_server
  allow_unverified_ssl  = var.vsphere_insecure
}

data "vsphere_datacenter" "dc" {
  name = var.vsphere_datacenter
}

data "vsphere_resource_pool" "pool" {
  name          = var.vsphere_resource_pool
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_host" "host" {
  name          = var.vsphere_host_system
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_datastore" "datastore" {
  name          = var.vsphere_datastore
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_network" "network" {
  name          = var.vsphere_network
  datacenter_id = data.vsphere_datacenter.dc.id
}

resource "vsphere_virtual_machine" "docker-hosts" {
  name             = "vsphere-provider-test01"
  resource_pool_id = data.vsphere_resource_pool.pool.id
  datastore_id     = data.vsphere_datastore.datastore.id
  datacenter_id    = data.vsphere_datacenter.dc.id
  host_system_id   = data.vsphere_host.host.id

  num_cpus = 4
  memory   = 4096
  guest_id = "other3xLinux64Guest"
  hv_mode  = "hvOff"

  disk {
    label = "disk0"
    size  = 20
  }

  network_interface {
    network_id = data.vsphere_network.network.id
  }

  ovf_deploy {
    remote_ovf_url       = var.vsphere_ovf_url
    disk_provisioning    = "thin"
    ovf_network_map = {
      "nat" = data.vsphere_network.network.id
    }
  }

  vapp {
    properties = {
      "guestinfo.hostname" = "vsphere-provider-test01",
      "guestinfo.dns1" = "8.8.8.8",
      "guestinfo.dns2" = "8.8.4.4",
      "guestinfo.nic1.name" = "ens33",
      "guestinfo.nic1.ipaddress" = "192.168.0.10",
      "guestinfo.nic1.netmask" = "255.255.255.0",
      "guestinfo.nic1.gateway" = "192.168.0.1",
    }
  }
}

```

### Debug Output

https://gist.github.com/blakewade/b9874daed1732d27008558b7f09d034a

### Expected Behavior

The OVF is deployed and the values for the vApp properties are set

### Actual Behavior

No values for the vApp properties have been set

![image](https://user-images.githubusercontent.com/71611767/96388248-10191c80-11eb-11eb-9873-2e9d04d1fc86.png)

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This is currently working for vSphere Provider 1.24.0

![image](https://user-images.githubusercontent.com/71611767/96388409-f3311900-11eb-11eb-9ac7-c18be9ad01f8.png)

* #0000

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

Hi, 

I'm trying to use your example and I'm stuck  referring to the OVF source. 
During the **apply**, when I use a remote URL (ex.: ) I got 

``` 
Error: the given ovf file https://192.168.1.10/folder/template%5fcicd%2ftst1%2dcicd%2dzaa%2d1/tst1%2dcicd%2dzaa%2d1%2eova?dcPath=LAB%252dVIRTU%252d01&dsName=esxlab0003%255fLocal is empty

```

Result with TF_LOG=Trace: 

```
vsphere_virtual_machine.docker-hosts: Creating...
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalApply
2020/12/09 16:24:27 [DEBUG] EvalApply: ProviderMeta config value set
2020/12/09 16:24:27 [DEBUG] vsphere_virtual_machine.docker-hosts: applying the planned Create change
2020/12/09 16:24:27 [TRACE] GRPCProvider: ApplyResourceChange
2020-12-09T16:24:27.035+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] setting computed for "guest_ip_addresses" from ComputedKeys
2020-12-09T16:24:27.035+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] setting computed for "vapp_transport" from ComputedKeys
2020-12-09T16:24:27.035+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] vsphere_virtual_machine (ID = <new resource>): Beginning create
2020-12-09T16:24:27.035+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] tagsClientIfDefined: No tags configured, skipping loading of tagging client
2020-12-09T16:24:27.041+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] VM is being deployed from ovf/ova template https://192.168.1.10/folder/template%5fcicd%2ftst1%2dcicd%2dzaa%2d1/tst1%2dcicd%2dzaa%2d1%2eova?dcPath=LAB%252dVIRTU%252d01&dsName=esxlab0003%255fLocal
2020-12-09T16:24:27.041+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Locating resource pool with ID resgroup-8
2020-12-09T16:24:27.055+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Resource pool found: resgroup-8
2020-12-09T16:24:27.056+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Locating folder at path "" relative to virtual machine root
2020-12-09T16:24:27.146+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Folder located: "/LAB-VIRTU-01/vm"
2020-12-09T16:24:27.146+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Locating host system ID host-24
2020-12-09T16:24:27.162+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Host system found: host-24
2020-12-09T16:24:27.163+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Locating datastore with ID "datastore-30"
2020-12-09T16:24:27.177+0100 [DEBUG] plugin.terraform-provider-vsphere_v1.24.2_x4: 2020/12/09 16:24:27 [DEBUG] Datastore with ID "datastore-30" found
2020/12/09 16:24:27 [DEBUG] vsphere_virtual_machine.docker-hosts: apply errored, but we're indicating that via the Error pointer rather than returning it: the given ovf file https://192.168.1.10/folder/template%5fcicd%2ftst1%2dcicd%2dzaa%2d1/tst1%2dcicd%2dzaa%2d1%2eova?dcPath=LAB%252dVIRTU%252d01&dsName=esxlab0003%255fLocal is empty
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalMaybeTainted
2020/12/09 16:24:27 [TRACE] EvalMaybeTainted: vsphere_virtual_machine.docker-hosts encountered an error during creation, so it is now marked as tainted
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalWriteState
2020/12/09 16:24:27 [TRACE] EvalWriteState: removing state object for vsphere_virtual_machine.docker-hosts
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalApplyProvisioners
2020/12/09 16:24:27 [TRACE] EvalApplyProvisioners: vsphere_virtual_machine.docker-hosts has no state, so skipping provisioners
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalMaybeTainted
2020/12/09 16:24:27 [TRACE] EvalMaybeTainted: vsphere_virtual_machine.docker-hosts encountered an error during creation, so it is now marked as tainted
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalWriteState
2020/12/09 16:24:27 [TRACE] EvalWriteState: removing state object for vsphere_virtual_machine.docker-hosts
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalIf
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalIf
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalWriteDiff
2020/12/09 16:24:27 [TRACE] eval: *terraform.EvalApplyPost
2020/12/09 16:24:27 [ERROR] eval: *terraform.EvalApplyPost, err: the given ovf file https://192.168.1.10/folder/template%5fcicd%2ftst1%2dcicd%2dzaa%2d1/tst1%2dcicd%2dzaa%2d1%2eova?dcPath=LAB%252dVIRTU%252d01&dsName=esxlab0003%255fLocal is empty
2020/12/09 16:24:27 [ERROR] eval: *terraform.EvalSequence, err: the given ovf file https://192.168.1.10/folder/template%5fcicd%2ftst1%2dcicd%2dzaa%2d1/tst1%2dcicd%2dzaa%2d1%2eova?dcPath=LAB%252dVIRTU%252d01&dsName=esxlab0003%255fLocal is empty
2020/12/09 16:24:27 [TRACE] [walkApply] Exiting eval tree: vsphere_virtual_machine.docker-hosts
2020/12/09 16:24:27 [TRACE] vertex "vsphere_virtual_machine.docker-hosts": visit complete
2020/12/09 16:24:27 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/12/09 16:24:27 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/vsphere\"] (close)" errored, so skipping
2020/12/09 16:24:27 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020/12/09 16:24:27 [TRACE] statemgr.Filesystem: not making a backup, because the new snapshot is identical to the old
2020/12/09 16:24:27 [TRACE] statemgr.Filesystem: no state changes since last snapshot
2020/12/09 16:24:27 [TRACE] statemgr.Filesystem: writing snapshot at terraform.tfstate

Error: the given ovf file https://192.168.1.10/folder/template%5fcicd%2ftst1%2dcicd%2dzaa%2d1/tst1%2dcicd%2dzaa%2d1%2eova?dcPath=LAB%252dVIRTU%252d01&dsName=esxlab0003%255fLocal is empty

  on main.tf line 104, in resource "vsphere_virtual_machine" "docker-hosts":
 104: resource "vsphere_virtual_machine" "docker-hosts" {


2020/12/09 16:24:27 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2020/12/09 16:24:27 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2020-12-09T16:24:27.392+0100 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/vsphere/1.24.2/linux_amd64/terraform-provider-vsphere_v1.24.2_x4 pid=6175
2020-12-09T16:24:27.393+0100 [DEBUG] plugin: plugin exited
```
have you ever seen this behavior?

Thanks.

--- a/vsphere/virtual_machine_config_structure.go
+++ b/vsphere/virtual_machine_config_structure.go
@@ -5,6 +5,8 @@ import (
 	"log"
 	"reflect"
 
+	"github.com/mitchellh/copystructure"
+
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
@@ -628,9 +630,11 @@ func expandVAppConfig(d *schema.ResourceData, client *govmomi.Client) (*types.Vm
 	if newVApps != nil && len(newVApps) > 0 && newVApps[0] != nil {
 		newVApp := newVApps[0].(map[string]interface{})
 		if props, ok := newVApp["properties"].(map[string]interface{}); ok {
-			for k, v := range props {
-				newMap[k] = v
+			propsCopy, err := copystructure.Copy(props)
+			if err != nil {
+				return nil, fmt.Errorf("while extracting vapp properties into a new map: %s", err)
 			}
+			newMap = propsCopy.(map[string]interface{})
 		}
 	}
 

--- a/vsphere/virtual_machine_config_structure.go
+++ b/vsphere/virtual_machine_config_structure.go
@@ -628,7 +628,9 @@ func expandVAppConfig(d *schema.ResourceData, client *govmomi.Client) (*types.Vm
 	if newVApps != nil && len(newVApps) > 0 && newVApps[0] != nil {
 		newVApp := newVApps[0].(map[string]interface{})
 		if props, ok := newVApp["properties"].(map[string]interface{}); ok {
-			newMap = props
+			for k, v := range props {
+				newMap[k] = v
+			}
 		}
 	}
 

